sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -590,7 +591,11 @@ object ScalaReflection extends ScalaReflection {
               "cannot be used as field name\n" + walkedTypePath.mkString("\n"))
           }
 
-          val fieldValue = Invoke(inputObject, fieldName, dataTypeFor(fieldType))
+          // primitive take only non-null or struct takes non-null object guarded by isNull
```

Comment:
ExpressionEncoder can't encode `None` at the root level, as Spark SQL doesn't allow top-level row to be null.